### ISSUE TITLE
Sorting broken when using WebUI

### ISSUE THREAD

**Issue by icenine9** [Carve/qbittorrent-webui-cjratliff.com#8]
Despite clearing cache, sorting appears to be broken. 
![image](https://user-images.githubusercontent.com/16182988/228982459-d4e8cc6a-6403-44f9-b8b8-13a20a0fb99a.png)
qbittorrent-nox 4.5.2 on ubuntu 22.10 (kernel 6.2.2 mainline). Note screenshot showing that sort is set to descending, yet percentages are obviously not in descending order.

**Comment by WalzingMatilda** [comment]
Same here. 
On Other columns, it  seems that I can sort in ascendant/descendant order without any problem... so it seems to be a progress_bar problem only.

**Comment by Carve** [comment]
> Same here. On Other columns, it seems that I can sort in ascendant/descendant order without any problem... so it seems to be a progress_bar problem only.

What OS and qbit ?

**Comment by Carve** [comment]
> Despite clearing cache, sorting appears to be broken. ![image](https://user-images.githubusercontent.com/16182988/228982459-d4e8cc6a-6403-44f9-b8b8-13a20a0fb99a.png) qbittorrent-nox 4.5.2 on ubuntu 22.10 (kernel 6.2.2 mainline). Note screenshot showing that sort is set to descending, yet percentages are obviously not in descending order.

Does this problem still occur without the use of my theme for you?
Do you have any errors in your browser console?

**Comment by icenine9** [comment]
No, problem only occurs when using your theme.

And there is one error on the console:

DevTools failed to load source map: Could not load content for chrome-extension://jpdapbcmfllbpojmkefcikllfeoahglb/content_start/bundle.js.map: System error: net::ERR_BLOCKED_BY_CLIENT

That error appears in other sites as well, every one it seems. I'm guessing it has to do with the bypass-paywall plugin that I have manually installed, so i'm not sure if it's contributing to this problem or not.

**Comment by Carve** [comment]
> No, problem only occurs when using your theme.
> 
> And there is one error on the console:
> 
> DevTools failed to load source map: Could not load content for chrome-extension://jpdapbcmfllbpojmkefcikllfeoahglb/content_start/bundle.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
> 
> That error appears in other sites as well, every one it seems. I'm guessing it has to do with the bypass-paywall plugin that I have manually installed, so i'm not sure if it's contributing to this problem or not.

ok, thanks for the additional info.  I will assume since it works with the extension and NOT using my theme then it isn't a problem with the extension, however.. if you want you could try to disable the extension and see if it sorting worked then, my guess will be probably not.

I haven't been able to replicate the issue, but I am setting up a Ubuntu 22 (deskop) today.  Are you running any special variant of Ubuntu or just the normal one with the normal window manager?

**Comment by icenine9** [comment]
Nothing special aside from using qbittorrent-nox which has no UI (web interface only)

**Comment by WalzingMatilda** [comment]
I have qbitorrent 4.5.2 (with gui) on a linux mint 21.3 server. 
I'm using WebUI via LAN on a ubuntu 20.04 and firefox

the only thing I can see on thebrowser console is this message:
  

_This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://firefox-source-docs.mozilla.org/performance/scroll-linked_effects.html for further details and to join the discussion on related tools and features!_

but I only get this one time in a random moment: refreshing page (ctrl + R) and sorting columns doesn't seem to produce anything strange on the browser console.

Then I tested it on Chromium;
The sorting problem is the same, but it complains  about the javascript framework you use:


    _A page or script is accessing at least one of navigator.userAgent, navigator.appVersion, and navigator.platform. 
Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.
     To fix this issue, replace the usage of navigator.userAgent, navigator.appVersion, and navigator.platform with feature detection, progressive enhancement, or migrate to navigator.userAgentData.
       Note that for performance reasons, only the first access to one of the properties is shown._

  _AFFECTED RESOURCES

  MooTools-Core-1.6.0-compat-compressed.js_

Hope this can help. 


more about this:
https://blog.chromium.org/2021/09/user-agent-reduction-origin-trial-and-dates.html

 
EDIT:

I just checked and I can say the the sorting problem on the progress bar happens on the main bar only.

if I select a torrent with numerous files inside it, then press "content" and try to order the files in ascending or descending order, it works perfectly. 

In other words I can't  sort torrents' progress bar, but I can sort perfectly files' progress bar

**Comment by Carve** [comment]
I've got Ubuntu 22 LTS installed and testing the webui and have not been able to replicate the issue at hand.  I'm determined to replicate this issue so I can get it fixed.

**Comment by WalzingMatilda** [comment]
I just tried a different approach:

ubuntu 20.04 
qbittorrent 4.3.9 (with gui) 
WebUi on localhost (just to simplify my setup)

Same thing appened. 

I think it has something to do with bubble sorting and decimal point:

when I sort torrents on 4.3.9 I have:

100.0
100.0
100.0
9.9
98.6
75.1

so perhaps when figures are only 2, it ignores the decimal point?


*******


on the other setup (the one I was talking about in my previous post), with different torrents and on a 4.5.2 release:

100.0
100.0
100.0
100.0
2.5
1.9
10.1
94.3
92.8
80.2
75.7
64.1
5.7
45.5
39.4
15.4
0.0

so, if I read 25 instead of 2.5, 19 instead of 1.9, I can spot a kind of sorting: 25 - 19 - 10.1;

same thing if I read 5.7 as 57;

The only mistery is why 25 19 10.1 are out of order.

Any chance you know where this figures can be manipulated? It would be interesting to have a format on 4 figures (for example: 001.0) just to see if the problem disappear.....

 
 
********
P.S.
my last experiment:
I try to access this setup directly on my mint 21.3 (I think is based on ubuntu 22) using firefox and localhost;
same exact order!!
I suppose you are using qbitorrent last release.... and it works on yours.... I don't understand.....

*********
P.S.

I just noticed that the only torrent currently in download (the one at 80.2 , now at 80.5), every now and then, with no mouse hovering of any kind and with no regular interval of time, jumps from its position and it goes down between 39.4 and 15.4 and then return to its original position: this behavior, without any change of the figures.


The last thing: if I sort the progress bar in the ascending position, it reverses perfectly  the order I showed you, and even the torrent in download jumps in the same way.....

**Comment by WalzingMatilda** [comment]
just saw the image icenine9 posted. 
Two main blocks of sorting..... 

the only one absolutely out of any logic seems to be 61.6.....
but if you move the decimal point, it would fit:  6.3 - 6.16 - 6.1........

**Comment by Carve** [comment]
@WalzingMatilda @icenine9 Just to confirm; you don't have this error with the default webUI theme?  (I ask because my theme is based on the default theme)

**Comment by icenine9** [comment]
Correct.

**Comment by WalzingMatilda** [comment]
No, that work as expected....

**Comment by Carve** [comment]
Thanks for the updates.  I'm working on the resolution for this

**Comment by Carve** [comment]
Fixed with today's update.  Sorry it took longer than anticipated

**Comment by icenine9** [comment]
Cmon now, share  what the problem was 😊

**Comment by Carve** [comment]
> Cmon now, share what the problem was 😊

The problem was in one of the Libraries (and a core change).  Updating the Libraries to the latest used by the src in qBittorrent and the JavaScript for the sorting and it now works as intended.  Nothing fancy really 🤣 

![image](https://github.com/Carve/qbittorrent-webui-cjratliff.com/assets/662856/9c39def7-9fdf-4404-90be-8810ab0a1011)

**Comment by icenine9** [comment]
Confirmed! Thanks so much for fixing this. It's now the perfect webui :)

**Comment by Carve** [comment]
> Confirmed! Thanks so much for fixing this. It's now the perfect webui :)

My pleasure!

**Comment by WalzingMatilda** [comment]
Confirmed! work for me too!
The only thing I noticed is that now the color parameters inside the progressbar.js file seems to be partially ignored: 
 I can't change the text color to white like in the previous version(I have a really annoying problem with display brightness, so I "darkened" your skin a bit more - file attached). Only if you know what's going on without having to test it yourself. 
It's good this way for me...

![Before](https://github.com/Carve/qbittorrent-webui-cjratliff.com/assets/129553000/27fd31ea-2cdf-4f87-8311-b711f5210497)

![Now](https://github.com/Carve/qbittorrent-webui-cjratliff.com/assets/129553000/27e46587-0567-456b-8a9c-aa9ef0f53b4d)
